apps/examples/rtc/rtc_main.c

```diff
@@ -0,0 +1,222 @@
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <errno.h>
+#include <pthread.h>
+#include <sched.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <time.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <tinyara/rtc.h>
+
+#define TEST_TIME 300	/* 300 secs */
+#define TEST_STACKSIZE 4096
+
+static volatile bool rtctest_started;
+
+static void rtc_test_stop(void)
+{
+	/* Has the rtc accuracy test already started? */
+	if (rtctest_started) {
+		/* Stop the test. */
+		printf("Stopping, not stopped yet.\n");
+		rtctest_started = FALSE;
+	} else {
+		printf("There is no active rtc accuracy test.\n");
+	}
+}
+
+static void *rtc_test(void *arg)
+{
+	int ret;
+	int fd;
+	struct rtc_time prev_time;
+	struct rtc_time next_time;
+	struct tm prev_tm;
+	struct tm next_tm;
+	time_t prev;
+	time_t next;
+	int total_cnt = 0;
+
+	printf("=== RTC Accuracy Test ===\n");
+
+	/* This sample runs forever with printing the log repeatedly on every TEST_TIME secs until "rtc stop".
+	 * You can check and compare the time which is from Serial Log Tool like Teraterm and the printed log from line 140.
+	 */
+
+	memset(&prev_time, 0, sizeof(struct rtc_time));
+	memset(&next_time, 0, sizeof(struct rtc_time));
+	memset(&prev_tm, 0, sizeof(struct tm));
+	memset(&next_tm, 0, sizeof(struct tm));
+
+	fd = open("/dev/rtc0", O_RDWR);
+	if (fd < 0) {
+		printf("ERROR : Fail to open rtc.\n");
+		return NULL;
+	}
+
+	/* Initialize the RTC time as 0 */
+	ret = ioctl(fd, RTC_SET_TIME, (unsigned long)&prev_time);
+	if (ret < 0) {
+		printf("ERROR : Fail to set RTC init time.\n");
+		close(fd);
+		return NULL;
+	}
+
+	sched_lock();
+	while (rtctest_started) {
```

Comment:
When disable_signal config is enabled, this function is called with this variable 0. Is it ok?

Reply:
Modified. Thanks.